stages/org.osbuild.ostree.config

```diff
@@ -74,6 +79,10 @@ def main(tree, options):
         ro = "true" if readonly else "false"
         ostree("config", "set", "sysroot.readonly", ro, repo=repo)
 
+    composefs = sysroot_options.get("composefs")
+    if composefs is not None:
+        ostree("config", "set", "ex-integrity.composefs", composefs, repo=repo)
```

Comment:
So it absolutely makes sense to start this but one problem is that as soon as a PR like this is merged it becomes 10x harder to change the interface in ostree.

Reply:
We could move this particular stage to some custom automotive specific stage. Would that help?

Reply:
cc @ondrejbudai

Reply:
The one thing we try to avoid is two runs of the same manifest resulting in different artefacts. So we should not quietly change the behaviour of a stage.

@cgwalters the APIs used here are released in RHEL/Fedora, so presumably they already have some sort of stability guarantee? Or is this special in some way?

Reply:
The intention of the `ex.` prefix in this config option is exactly that it is experimental and subject to change.

Reply:
Does that mean it should only be used for development, and not as part of shipped products?

Reply:
It's perhaps like Tech Preview in RHEL terms, or a nightly-only feature in Rust etc.  We need to "ship" it so it's easier to try and gain real world feedback, but not commit to supporting the exact layout and names of config options ~forever.

Reply:
Can we maybe mirror this by calling the osbuild option `experimental-composefs`, as that is what it does, it enables an experimental feature. It will forever in the future enable an experimental feature, but you then get what you asked for.

Reply:
One the feature stabilizes in the fututre, we can also add a regular `composefs` option.

Reply:
Might be useful to have something mirroring this behaviour, yes. Mark something as experimental and allow it to be dropped / renamed in the future. The only thing we want to avoid is magically changing behaviour, but "experimental" manifests no longer building might be ok.

Should be up to @ondrejbudai and team though.

Reply:
This is a whole policy change we should all agree on it seems.  I'm not entirely against it.  One thing I would say for the record is that we should never put these experimental features in other projects (osbuild/images and osbuild-composer).

In terms of what it should look like and how it should work: My first instinct is to isolate things like these into separate objects in the stage schema
```json
"properties": {
  "config": {
    "type": "object",
    "additionalProperties": false,
    "description": "OSTree configuration groups",
    ...
  },
  "experimental": {
    "composefs": {
      ...
    }
  }
}
```

(or maybe `org.osbuild.experimental` to avoid possible name collisions)
and maybe generating a warning when building a manifest that contains an experimental option.  We could even go as far as requiring a command line flag to enable it, but maybe this is a bit extreme.